Update kin-openapi package with changes related to request validation

nopcoder opened the pull request:
Close https://github.com/treeverse/lakeFS/issues/9168

Check the issue for information about how the upgrade changes the code in terms of validation and behaviour.

arielshaqed commented:
Does it also fix #8474 ?  Can/should we do both at once?

arielshaqed reviewed (changes requested):
Thanks!  Seeking clarity about why we needed to make some changes in (I _think_) the lakeFS client we provide to for.

Also tests might be more passing before we go ahead with this 😝

nopcoder commented on api/swagger.yml:
```diff
@@ -43,6 +43,7 @@ components:
     PaginationPrefix:
       in: query
       name: prefix
+      allowEmptyValue: true
```
1. Test failed on this one
2. The code supported empty value, in the spec we didn't specified it and the validator didn't enforce it. As we upgrade, so we got a lot of bug fixes and features - this specific one [v0.93.0](https://github.com/getkin/kin-openapi/releases/tag/v0.93.0)

nopcoder commented on api/swagger.yml:
```diff
@@ -286,12 +290,12 @@ components:
           type: boolean
-          example: "true"
           default: false
+          example: true
```
This one was found because we parse the spec as we init our middlewares

yonipeleg33 commented:
Nothing to add on top of @arielshaqed's (awesome) review. Thanks! :eyes:

nopcoder commented on contrib/auth/apigen/authapi.gen.go:
this package changes things are runtime, so you will not find anything interesting in the generated code.

nopcoder commented on go.sum:
```diff
@@ -463,9 +465,11 @@ github.com/go-logr/stdr v1.2.2/go.mod h1:mMo/vtBO5dYbehREoey6XUKy/eSumjCCveDpRre
 github.com/go-openapi/jsonpointer v0.19.5/go.mod h1:Pl9vOtqEWErmShwVjC8pYs9cog34VGT37dQOVbmoatg=
+github.com/go-openapi/jsonpointer v0.21.0/go.mod h1:IUyH9l/+uyhIYQ/PXVA41Rexl+kOkAPDdXEYns6fzUY=
 github.com/go-openapi/swag v0.19.5/go.mod h1:POnQmlKehdgb5mhVOsnJFsivZCEZ/vjK9gh66Z9tfKk=
 github.com/go-openapi/swag v0.19.14 h1:gm3vOOXfiuw5i9p5N9xJvfjvuofpyvLA9Wr6QfK5Fng=
 github.com/go-openapi/swag v0.19.14/go.mod h1:QYRuS/SOXUCsnplDa677K7+DxSOj6IPNl/eQntq43wQ=
+github.com/go-openapi/swag v0.23.0/go.mod h1:esZ8ITTYEsH1V2trKHjAN8Ai7xHb8RV+YSZ577vPjgQ=
```
Thanks - updated tidy + sync

nopcoder commented on pkg/actions/lua_test.go:
```diff
@@ -528,7 +528,7 @@ print(code, resp.id, resp.commit_id)
 			ExpectedRequest: map[string]any{
 				"repository": "repo",
-				"body":       apigen.CreateTagJSONRequestBody{Id: "tag1", Ref: "main"},
+				"body":       apigen.CreateTagJSONRequestBody{Id: "tag1", Ref: "main", Force: apiutil.Ptr(false)},
```
This is what lakeFS server will process now - the test match the expected to get on the server side based on recording.
In this case because 'force' had default:false it will be rendered vs previous previous version where if the client didn't post anything we will not see the parameter.

nopcoder commented on pkg/actions/lua_test.go:
```diff
@@ -556,6 +556,7 @@ print(code, resp)
 					Amount:    apiutil.Ptr[apigen.PaginationAmount](100),
 					Prefix:    apiutil.Ptr[apigen.PaginationPrefix]("prefix"),
 					Delimiter: apiutil.Ptr[apigen.PaginationDelimiter]("delim"),
+					Type:      apiutil.Ptr("three_dot"),
```
same here - we specified the default value on this parameter.

nopcoder commented:
@arielshaqed thanks for the quick review, updated all the parameters that we accepts empty string, exclude the one we set as required and addressed all the comments.

arielshaqed reviewed (approved):
Great, thanks!

This is a pretty big `include-changelog`.  For instance, _if_ there was any inconsistency in "default" request parameters, then _now_ users will actually see that difference.  That effectively ends up fixing a bug, so cool... but users might see a difference.
